src/pages/ConnectWallet/MobileConnect.tsx

```diff
@@ -198,19 +209,23 @@ export const MobileConnect = () => {
         <BodyStack>
           <Stack textAlign='center' spacing={2}>
             <Heading>{translate('connectWalletPage.welcomeBack')}</Heading>
-            <BodyText>{translate('connectWalletPage.mobileSelectBody')}</BodyText>
+            {!isWaitingForRedirection && !state.isLoadingLocalWallet ? (
+              <BodyText>{translate('connectWalletPage.mobileSelectBody')}</BodyText>
+            ) : null}
           </Stack>
           <Stack>
-            {isWaitingForRedirection ? (
+            {isWaitingForRedirection || state.isLoadingLocalWallet ? (
               <Center py={6}>
                 <Spinner />
               </Center>
             ) : (
-              <MobileWalletList onIsWaitingForRedirection={handleIsWaitingForRedirection} />
+              <>
+                <MobileWalletList onIsWaitingForRedirection={handleIsWaitingForRedirection} />
+                <Button size='lg-multiline' variant='outline' onClick={handleToggleWallets}>
+                  {translate('connectWalletPage.createOrImport')}
+                </Button>
```

Comment:
No reason to display the import button if we are doing things with the wallet state, it should be safe to use only if the WalletProvider isn't doing something under the hood